### ISSUE TITLE
How to handle inference on arrays

### ISSUE THREAD

**Issue by jkelleyrtp** [microsoft/SandDance#234]
I have a field in my data

```javascript
position = [0, 0, 0];
```

I'd like to have three columns from this field

```javascript
posx = position[0];
posy = position[0];
posz = position[0];
```

Can I do this without modifying the underlying array?

**Comment by danmarshall** [comment]
Yes, we have an experimental feature for "calculated columns". I've just added a link to our [examples page](https://sanddance.js.org/examples/) showing an example which [splits a string into a new column](https://sanddance.js.org/tests/v3/umd/transforms.html).

On the insight object there is a `transform` which takes an array of [Vega transforms](https://vega.github.io/vega/docs/transforms/). In our example we use the [formula transform](https://vega.github.io/vega/docs/transforms/formula/) which creates a new column based on Vega's [expression language](https://vega.github.io/vega/docs/expressions/). 

In the future, we'd like to build UI for adding calculated columns based on Vega's transformers.
